std/root.ln

```diff
@@ -631,6 +631,154 @@ export fn toHashMap(kva: Array<KeyVal<Hashable, any>>) {
   return hm
 }
 
+// Tree implementation
+export type Tree<T> {
+  vals: Array<T>
+  parents: Array<int64>
+  children: Array<Array<int64>>
+}
+
+export type Node<T> {
+  id: int64
+  tree: Tree<T>
+}
+
+export fn newTree(rootVal: any): Tree<any> = new Tree<any> {
+  vals = new Array<any> [ rootVal ]
+  parents = new Array<int64> [ -1 ]
```

Comment:
nit - can we have a few short comments on how the initialization to -1 works and how `Node` and `Tree` relate to each other?